File: tailoring-online-services/order-management-service/src/main/java/com/order_management_service/dto/CustomizedProductDto.java
package com.order_management_service.dto;

import com.order_management_service.model.Product;
import lombok.*;

import java.io.Serializable;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class CustomizedProductDto implements Serializable {
    private Product product;
    private List<CustomizedMeasurementDto> measurementDtos;
    private List<CustomizedOptionDto> optionDtos;
}

File: tailoring-online-interfaces/src/app/core/dtos/customized-product.dto.ts
import { Product } from "../interfaces/product.interface";
import { CustomizedMeasurementDto } from "./customized-measurement.dto";
import { CustomizedOptionDto } from "./customized-option.dto";

export interface CustomizedProductDto {
  product: Product;
  measurementDtos: CustomizedMeasurementDto[];
  optionDtos: CustomizedOptionDto[];
}
